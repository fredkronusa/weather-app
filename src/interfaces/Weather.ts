export interface IWeeklyForecastResponse {
  city_name: string;
  data: IForecastData[];
}

export interface IWeeklyForecast {
  weeklyForecast: IForecastData[];
}

export interface IWeather {
  description: string;
  icon: string;
}

export interface IForecastData {
  max_temp: number;
  min_temp: number;
  uv: number;
  valid_date: string;
  weather: IWeather;
}

export interface ICurrentForecastResponse {
  city_name: string;
  weather: IWeather;
  precip: number;
  rh: number;
  wind_spd: number;
  wind_cdir: string;
  ob_time: string;
  temp: number;
  uv: number;
}

export interface ICurrentForecast {
  city: string;
  weather: IWeather;
  precipitation: string;
  humidity: string;
  wind: string;
  date: string;
  temperature: number;
  uv: string;
}
