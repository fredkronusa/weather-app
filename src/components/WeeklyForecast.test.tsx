import * as React from "react";

import { ICurrentForecast, IWeeklyForecast } from "../interfaces/Weather";

import WeeklyForecastComponent from "./weeklyForecast";
import { render } from "@testing-library/react";

const currentForecastMockData: IWeeklyForecast[] = [
  {
    max_temp: 15,
    min_temp: 12,
    uv: 3,
    valid_date: "2020-08-15",
    weather: {
      icon: "123",
      description: "foggy",
    },
  },
  {
    max_temp: 15,
    min_temp: 12,
    uv: 3,
    valid_date: "2020-08-15",
    weather: {
      icon: "123",
      description: "foggy",
    },
  },
];

describe("<WeeklyForecastComponent />", () => {
  // SNAPSHOT
  it("should match snapshot", async () => {
    const { asFragment } = render(
      <WeeklyForecastComponent
        weeklyForecast={currentForecastMockData}
        isCelsius={true}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
