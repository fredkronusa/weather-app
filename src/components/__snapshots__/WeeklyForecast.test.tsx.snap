// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeeklyForecastComponent /> should match snapshot 1`] = `
<DocumentFragment>
  <ul
    aria-label="weekly weather"
    class="sc-AxjAm kFRSvt"
  >
    <li
      class="sc-AxirZ ePBdqv"
    >
      <p
        class="sc-AxiKw eUIEFs"
      >
        Today
      </p>
      <img
        alt="foggy"
        class="sc-AxheI gXEPjd"
        src="/images/icons/123.png"
      />
      <span
        aria-label="Max Temp"
        class="sc-AxhCb zmOJK"
      >
        15째
      </span>
      <span
        aria-label="Min Temp"
        class="sc-AxhCb sc-AxhUy dLvNBx"
      >
        12째
      </span>
      <span
        aria-label="UX Index"
        class="sc-AxgMl gtQttU"
      >
        UV Index 3
      </span>
    </li>
    <li
      class="sc-AxirZ ePBdqv"
    >
      <p
        class="sc-AxiKw eUIEFs"
      >
        Saturday
      </p>
      <img
        alt="foggy"
        class="sc-AxheI gXEPjd"
        src="/images/icons/123.png"
      />
      <span
        aria-label="Max Temp"
        class="sc-AxhCb zmOJK"
      >
        15째
      </span>
      <span
        aria-label="Min Temp"
        class="sc-AxhCb sc-AxhUy dLvNBx"
      >
        12째
      </span>
      <span
        aria-label="UX Index"
        class="sc-AxgMl gtQttU"
      >
        UV Index 3
      </span>
    </li>
  </ul>
</DocumentFragment>
`;
